x/pki/types/message_add_pki_revocation_distribution_point_test.go

```diff
@@ -0,0 +1,488 @@
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+			},
+			err: pkitypes.ErrNonRootCertificateSelfSigned,
+		},
+		{
+			name: "IsPAA true, certificate is non-root",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.IntermediateCertPem,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+			},
+			err: pkitypes.ErrRootCertificateIsNotSelfSigned,
+		},
+		{
+			name: "PAA is true, CRL signer certificate contains vid != msg.vid",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.RootCertWithPidVidInSubject,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+			},
+			err: pkitypes.ErrCRLSignerCertificateVidNotEqualMsgVid,
+		},
+		{
+			name: "PAA is false, cert does not contain vid",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &false_,
+				CrlSignerCertificate: testconstants.IntermediateCertPem,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+			},
+			err: pkitypes.ErrVidNotFound,
+		},
+		{
+			name: "PAA is false, cert pid provided, msg pid does not provided",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &false_,
+				CrlSignerCertificate: testconstants.NonRootCertWithPidVidInSubject,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+			},
+			err: pkitypes.ErrPidNotFound,
+		},
+		{
+			name: "PAA is false, cert pid != msg.pid",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &false_,
+				CrlSignerCertificate: testconstants.NonRootCertWithPidVidInSubject,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				Pid:                  1,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+			},
+			err: pkitypes.ErrCRLSignerCertificatePidNotEqualMsgPid,
+		},
+	}
+
+	positiveTests := []struct {
```

Comment:
Please have the following positive tests:
-  when vid is encoded in subject as Mvid
-  when vid is encoded in subject as OID value
-  when pid is encoded in subject as Mpid
-  when pid is encoded in subject as OID value